### PR TITLE
[BUG] Fixed a changed behavior of MOEX for shares data.

### LINKED ISSUE #734: MOEX Data Errors
I am getting data errors using pandas_datareader for the Moscow Exchange.  The attached PDF outlines the problem, which seems to have started after changes somewhere around 23 Nov 2019 when different fields started to be returns and the price errors appears.  I am on python 3.7 in the Anaconda distribution's Spyder console.  Prices returned in python do not agree with those shown on the MOEX web site.  This used to work very well until last week.

my python code:

import datetime
import pandas_datareader
sdate = datetime.datetime(2019,10,31)
edate = datetime.datetime.today()
code = 'LSRG'
data = pandas_datareader.DataReader(code, 'moex', start=sdate, end=edate)
print(data[['OPEN', 'HIGH', 'LOW', 'CLOSE']].tail(10))
data[['OPEN', 'HIGH', 'LOW', 'CLOSE']].plot()

output from the console:

TRADEDATE  OPEN    HIGH    LOW   CLOSE
2019-11-18  771.40  926.16  732.8  924.00
2019-11-19  922.32  922.32  731.2  916.80
2019-11-20  913.68  919.44  723.8  919.44
2019-11-21  913.68  913.68  722.8  913.20
2019-11-22  763.40  916.08  725.4  762.00
2019-11-25  767.20  920.64  729.2  764.80
2019-11-26  764.40  917.28  726.0  757.20
2019-11-27  759.80  911.76  722.0  760.60
2019-11-28  907.20  911.76  719.8  911.76
2019-11-29  910.32  910.32  719.4  759.40
￼
image

[MOEX Data Problem.pdf](https://github.com/pydata/pandas-datareader/files/3908118/MOEX.Data.Problem.pdf)

Is this a genuine problem or am I doing something stupid?

### PR DESCRIPTION
**Pull request by ikonnikov**
[+] Added a test for fixed functionality.

- [x] closes #734 
- [x] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] passes `black --check pandas_datareader`
- [ ] added entry to docs/source/whatsnew/vLATEST.txt

### CODE REVIEW

File: pandas_datareader/moex.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import pandas as pd
 
 from pandas_datareader.base import _DailyBaseReader
-from pandas_datareader.compat import StringIO, binary_type, concat, is_list_like
+from pandas_datareader.compat import StringIO, binary_type, concat, is_list_like, PY3
```
Comment (bashtage) [inline pandas_datareader/moex.py]: PDR is python 3 only, so no need to import Python 3 or provide a path for Python 2.7.

File: pandas_datareader/moex.py [hunk 2]
```diff
@@ -60,17 +60,16 @@ def __init__(self, *args, **kwargs):
...
                 "of read() or _get_metadata() methods"
             )
 
         return [
             self.__url_data.format(
-                engine=self.__engines[s], market=self.__markets[s], symbol=s
-            )
-            for s in self.symbols
+                engine=engine, market=market, symbol=s
+            ) for s in self.symbols if s in self.__markets_n_engines for market, engine in self.__markets_n_engines[s]
```
Comment (bashtage) [inline pandas_datareader/moex.py]: This line is too long. Probably avoid the comprehension.

File: pandas_datareader/moex.py [hunk 3]
```diff
@@ -172,12 +179,20 @@ def read(self):
...
+                    else:
+                        str_io = StringIO("\r\n".join(out_list).encode('utf-8'))
+
+                    dfs.append(self._read_lines(str_io))  # add a new DataFrame
         finally:
             self.close()
 
-        if len(dfs) > 1:
+        if len(dfs) == 0:
+            raise IOError("{} returned no data; check URL for invalid or correct a date interval".format(self.__class__.__name__))
```
Comment (bashtage) [inline pandas_datareader/moex.py]: Too long as well.

File: pandas_datareader/moex.py (hunk 1 above)
Comment (ikonnikov) [reply pandas_datareader/moex.py]: ok, deleted py27 backward compatibility.

File: pandas_datareader/moex.py (hunk 3 above)
Comment (ikonnikov) [reply pandas_datareader/moex.py]: Collapsed.

File: pandas_datareader/moex.py (hunk 2 above)
Comment (ikonnikov) [reply pandas_datareader/moex.py]: I split out a comprehension to next line. Comprehension is cosily.

**Comment by ikonnikov** [comment]
> LGMT.

Hello! What could I do next for merge my commits to master?

**Comment by bashtage** [comment]
The test you added is failing.  Any reason why?

**Comment by bashtage** [comment]
Nevermind - my fault.  Thanks for the contribution.

**Pull request merged**